\documentclass[12pt,a4paper]{article}
\usepackage[margin=2.5cm]{geometry}
\usepackage{amsmath,amssymb,graphicx,setspace,palatino,float,subcaption, booktabs, titlesec}
\usepackage[colorlinks, allcolors = blue]{hyperref}
\usepackage{natbib}
\citestyle{egu}

\captionsetup[figure]{labelfont=bf,textfont={small,normalfont}}
\captionsetup[subfigure]{labelsep=quad, labelfont={bf,small}, textfont=small, singlelinecheck=off, justification=raggedright, position=top}
\renewcommand{\thesubfigure}{\Alph{subfigure}}

\renewcommand{\subsectionautorefname}{Section}
\renewcommand{\subsubsectionautorefname}{Section}
\titleformat{\paragraph}[hang]{\normalfont\normalsize\itshape}{\theparagraph}{0em}{}
\titlespacing*{\paragraph}{0pt}{1\baselineskip}{3pt}
\setlength{\parindent}{0pt}

\renewenvironment{abstract}{%
  \normalsize
  \textbf{\large\abstractname\\[1em]} % with a normal space
}

\title{Neutral models of \textit{de novo} gene emergence show that gene evolution has a preferred trajectory}
\author{Bharat Ravi Iyengar, Erich Bornberg-Bauer}
\date{\small Institute for Evolution and Biodiversity, Westfalian Wilhelms -- University of M\"{u}nster, H\"{u}fferstrasse 1, 48149 M\"{u}nster, Germany}

\begin{document}

\onehalfspacing

\setlength{\abovedisplayskip}{0pt}
\setlength{\belowdisplayskip}{1em}

\maketitle

\begin{abstract}
New protein coding genes can emerge from genomic regions that previously did not contain any genes, via a process called \textit{de novo} gene emergence. To synthesize a protein, DNA must be transcribed as well as translated, both of which need certain DNA sequence features. Stable transcription requires promoters and polydenylation signal, while translation requires at least an open reading frame (ORF). We develop mathematical models based on mutation probabilities, and the assumption of neutral evolution, to find out how quickly genes emerge and are lost. We also investigate whether different DNA features evolve in a sequential manner, and if sequence composition is biased by mutation rate. We find out that genes are lost much more rapidly than they emerge, and genes with long ORFs preferentially arise in regions that are already transcribed. Our study not only answers some fundamental questions on the topic of \textit{de novo} emergence, but also provides a modeling framework for future studies.
\end{abstract}

\section{Introduction}

Organisms evolve new traits by expanding their functional genome. Evolution of new genes is one of the ways by which new traits can emerge. The definition of a gene is complicated, and has been changing constantly \citep{genepostencode}. We use the following working definition of a gene: a gene is a region of the genome that gives rise to a functional product, that is an RNA or a protein. The next big challenge is to define what a function is, and how to classify gene functions. One of the easiest definition of a gene product's function is its ability to improve the survival of the organism under one or more environmental conditions. Proteins perform the most diverse kinds of molecular functions ranging from catalysis of biochemical reactions to formation of cellular structures \citep{berg}, whereas RNAs that do not encode proteins like mRNAs, or participate in protein synthesis like rRNAs and tRNAs, are mostly involved in regulation of gene expression \citep{fncel,Statello2021}.

In this study we focus on the evolution of genes that encode new proteins. New protein coding genes frequently arise via duplication of existing protein coding genes. These duplicated genes then genetically and functionally diverge from their parent genes by accumulating mutations \citep{Long2003,IAD}. Recently it has been shown that new protein coding genes can arise independent of gene duplication, from genomic sequences that did not previously encode any protein. This phenomenon is called \textit{de novo} gene emergence \citep{EBB-F1000,vanOss}. \textit{De novo} protein coding genes can emerge from intergenic sequences, non-coding RNA genes, introns, and even regions that partially overlap with existing protein coding genes \citep{vanOss}. To express a protein, a genomic sequence should be transcribed as well as translated. Therefore, the gene needs sequence features that enable both these processes. 

The primary requirement for transcription is recruitment of RNA polymerase at the gene's promoter \citep{Promoters}. A part of the promoter, called the core promoter is the region that determines the start of transcription \citep{corepromoters}. The second requirement for a productive transcription is its termination at appropriate sites. Even though transcription is widespread in eukaryotic genomes, a phenomenon known as pervasive transcription, most RNAs are quickly eliminated by RNA degrading enzymes \citep{RNAturnover}. Most eukaryotic mRNAs and long non-coding RNAs, are polyadenylated at their 3' termini, which makes them resistant to degradation. Polyadenylation, which also marks the end of transcription \citep{polyAterm}, is facilitated by sequences known as polyadenylation signals. In prokaryotes, end of transcription is determined by sequences known as terminators \citep{ProkTerm}. Thus the synthesis of a stable RNA, requires promoters, as well as termination/polyadenylation signal sequences.

The second major requirement for protein expression is translation of the transcribed mRNA. The most fundamental requirement for translation is the presence of an open reading frame (ORF). Usually, an mRNA needs additional features to initiate protein synthesis. These features include ribosome binding sites in prokaryotes \citep{RBS}, and Kozak consensus sequences in eukaryotes \citep{kozak,kozakDroso,kozakHuman}. 

Although gain of transcription and translation features guarantees \textit{de novo} birth of a protein coding gene, it does not ensure that the gene would persist in the genome for many generations. The newly born gene can lose the features as easily as it gained them unless it is fixed in the genome via evolutionary selection. If the protein synthesized by the \textit{de novo} gene provides some fitness advantage to the host organism then it will undergo positive selection. Simulataneously, the protein should have low toxicity and cost of synthesis, to survive purifying selection. A common mechanism by which protein mediated toxicity occurs is by misfolding and aggregation of proteins \citep{misfold1,misfold2}.

In this study we develop probabilistic models of \textit{de novo} gene emergence. We focus on \textit{de novo} genes emerging from intergenic sequences, also known as ``protogenes'' \citep{protogenes}. We model \textit{de novo} gene emergence as a two step process. Specifically, the protogene can first gain transcription features and produce an untranslated RNA, which in turn acquires translation features subsequently. Alternatively, the protogene can first gain translational features, and later gain transcription. Our models are based on neutral evolution where natural selection does not affect the rate of gene emergence, or loss. Using these models, we predict that \textit{de novo} gene emergence has a preferred evolutionary trajectory, even under the assumption of neutrality.

\section{Results}

In this work, we developed mathematical models to estimate the rates and probabilities of \textit{de novo} gene emergence, as well as gene loss. A \textit{de novo} gene emerges from non-genic DNA, when the latter mutates to gain sequence features necessary for transcription and translation. Both transcription and translation are complex processes involving many biomolecular complexes that work in concert. Here, we focus on the minimal requirements for these processes to occur. Specifically for transcription, we require a gene to have one of the two core promoter motifs, the TATA-box or the Initiator element (Inr), and a polyadenylation (poly-A) signal sequence. We chose these two promoter motifs because they are the most abundant type of core promoters across diverse eukaryotes \citep{Promoters,corepromoters}. We defined the poly-A signal as a transcription requirement because it is crucial for transcript stability and its export to cytoplasm \citep{polyAterm,polyAexp}. For translation, we only require the gene to have an open reading frame (ORF). 

Using simple probability models, we calculate the likelihood of finding these features in a DNA sequence by random chance, and the rate at which these features are gained and lost via mutagenesis. These probability models are essentially described by two kinds of probability. First is the probability of finding a DNA sequence feature (such as TATA box or an ORF; \autoref{methprob}). This probability depends on the nucleotide distribution given by the GC-content. The second kind of probability is the transition probability, that is the probability of a DNA sequence mutating to another (\autoref{methgain}). Transition probabilities that describe the gain and loss of DNA sequence features, depend on mutation rate, mutation bias, and GC-content. 

For all calculations we used a GC-content of 0.42 which is reasonably close to the global GC-content of human, \textit{Drosophila melanogaster}, and \textit{Saccharomyces cerevisiae} genomes \citep{allgccontents, bionumbers}. We used a mutation rate of $7.8\times10^{-9}$ mutations per nucleotide position per generation, which corresponds to spontaneous mutation rate estimated in a \textit{Drosophila melanogaster} population that was subjected to periodic population bottlenecks \cite[mutation accumulation line,][]{drosophilamutrate}. We used mutation bias data (\autoref{mutbias}) from the same study \citep{drosophilamutrate} which was in agreement with another previous study on human pseudogenes \citep{humanmutrate}.

Using a similar approach, we predict how the nucleotide distribution and random mutations shape protein composition and evolution. 

Overall our models define a null hypothesis under which protein coding DNA sequences evolve neutrally through mutation pressure alone.

\subsection{How many times can a gene be lost?}

If a gene is found in one species but not in sister taxa, then the most common assumption is that this gene emerged only in one specific species. We asked what is the chance that the gene was present in a common ancestor, but was lost in all lineages except one. To this end we calculated the probability of gene emergence and gene loss (Materials and Methods). Specifically, the probability of gene gain can be defined as the sum of three probabilities. First is the probability that an ORF already exists, and a mutation causes transcription to emerge ($P_\textit{RNA-gain}$, \autoref{eqrnagain}) but doesn't disrupt the ORF ($P_\textit{ORF-stay}$, \autoref{eqorfstay}). Second is the probability that the DNA is transcribed, and an ORF emerges due to mutations ($P_\textit{ORF-gain}$, \autoref{eqorfgain}) while transcription stays intact ($P_\textit{RNA-stay}$, \autoref{eqrnastay}). Third is the probability that neither of the two features already exist and both emerge at the same time due to mutations (this probability is very small and is negligible). The following equation describes the probability of gene gain:

\begin{equation}
P_\textit{gene-gain} = P_\textit{ORF-stay}\times P_\textit{RNA-gain} + P_\textit{RNA-stay}\times P_\textit{ORF-gain} + P_\textit{RNA-gain}\times P_\textit{ORF-gain}
\label{genegaineq}
\end{equation}

Likewise, the probability of gene loss, given the gene (transcription and ORF) is already present, is the sum of the probabilities that transcription ($P_\textit{RNA-loss}$, \autoref{eqrnaloss}) or the ORF is lost due to mutations ($P_\textit{ORF-loss}$, \autoref{eqorfloss}). Specifically:

\begin{equation}
P_\textit{gene-loss} = P_\textit{RNA-loss} + P_\textit{ORF-loss}
\label{genelosseq}
\end{equation}

\begin{figure}[!t]
\centering
\includegraphics[scale=0.6]{Figures/geneGainLoss.pdf}
\caption{Genes are more likely to be independently lost twice, than being born once. The vertical axis shows the number of independent gene loss events (\autoref{genelosseq}) than can happen relative to one gene gain event (\autoref{genegaineq}) i.e. $\log(P_\textit{gene-gain})/\log(P_\textit{gene-loss})$. The horizontal axis shows the number of codons in the ORF. \autoref{genegaineq} describes total gene gain probability. Gene gain probability conditioned on the gene being absent would be $P_\textit{gene-gain}/(1-P_\textit{ORF} - P_\textit{RNA})$. Using the conditional probability of gene gain does not change these results.}
\label{gainlossprob}
\vspace{1ex}
\hrule
\end{figure}

The probability that a gene is lost $n$ times independently, is $(P_\textit{gene-loss})^{n}$. To find out how many independent gene loss events are as likely as a single gene gain event, we calculate the ratio of logarithms of gene gain and gene loss: $\log(P_\textit{gene-gain})/\log(P_\textit{gene-loss})$. For example, the ratio value of 2 would indicate that two independent gene loss events are as likely as a single gene gain event. We calculated this ratio for genes with different ORF lengths, because ORF gain and loss probabilities depend on the length of the ORF. We found that for all ORF lengths, the gene can be lost at least two times independently, in the time required to gain it (\autoref{gainlossprob}). 

We performed the same analysis on just the ORFs (and not the whole genes). That is, we calculated the ratio of logarithms of ORF gain and ORF loss. We found that for ORFs longer than 143 codons, the likelihood of two independent ORF loss events is more than or equal to a single ORF gain event. 

Overall, our analysis suggests that a \textit{de novo} gene expressed in only one species may not necessarily mean that it emerged for the first time in this species. 

\subsection{Does \textit{de novo} gene emergence have a preferred trajectory?}

For a \textit{de novo} gene to emerge from a non-genic DNA sequence, both transcription and ORF need to emerge. That is, probability of gene emergence is equal to the product of probabilities of transcription gain and ORF gain. Thus it may appear that the order of the occurrence of these two events does not matter. However, gene emergence is much more likely when one of the two features already exist (\autoref{genegaineq}) and therefore it has two possible trajectories -- ORF emerges first (ORF-first) or transcription emerges first (RNA-first). To understand whether \textit{de novo} gene emergence has a preferred trajectory, we calculated two probabilities. First, the probability that ORF exists and  mutations cause gain of transcription but no disruption of the ORF. This probability denotes the trajectory where ORF emerges first ($P_\textit{ORF-first}$, \autoref{orffirsteq1}). 

\begin{equation}
P_\textit{ORF-first} = P_\textit{ORF-stay}\times P_\textit{RNA-gain}
\label{orffirsteq1}
\end{equation}

The second probability ($P_\textit{RNA-first}$, \autoref{rnafirsteq1}) that denotes the trajectory where transcription emerges first, is the probability that DNA is already transcribed and mutations cause a gain of ORF but do not eliminate transcription. 

\begin{equation}
P_\textit{RNA-first} = P_\textit{RNA-stay}\times P_\textit{ORF-gain}
\label{rnafirsteq1}
\end{equation}

\begin{figure}[!t]
\centering
%\begin{subfigure}{0.48\textwidth}
%\centering
%\caption{One-step}
\includegraphics[scale=0.6]{Figures/first_ORF_RNA.pdf}
%\end{subfigure}\hfill%
%\begin{subfigure}{0.48\textwidth}
%\centering
%\caption{Two-step}
%\includegraphics[scale=0.6]{Figures/first_ORF_RNA2.pdf}
%\end{subfigure}
\caption{\textit{De novo} genes with shorter ORFs ($<$ 42 codons) preferentially emerge from existing ORFs (ORF-first) whereas \textit{de novo} genes with longer ORFs preferentially emerge from existing RNAs (RNA-first). The horizontal axis denotes the number of codons in the ORF. The vertical axis shows the log transformed ratio of single-step probabilities of ORF-first and RNA-first trajectories of \textit{de novo} emergence (Equations \ref{orffirsteq1} \& \ref{rnafirsteq1}). A positive value suggests an ORF-first trajectory and a negative value supports an RNA-first trajectory. Similar analysis with two-step probabilities shows a smaller difference between the likelihood of the two trajectories (Equations \ref{orffirsteq2} \& \ref{rnafirsteq2}).}
\label{whoisfirst}
\vspace{1ex}
\hrule
\end{figure}
We calculated the log-transformed ratio of $P_\textit{ORF-first}$ and $P_\textit{RNA-first}$, such that a positive value of the ratio would mean that transcription gain for an untranscribed ORF (ORF-first) is more likely than ORF gain in an already transcribed DNA (RNA-first). In other words, the ORF-first hypothesis is more feasible. Likewise, a negative value of the ratio would suggest that the RNA-first hypothesis is more feasible. With the specific parameter values we used, it turns out that \textit{de novo} genes with up to 42 codons in their coding region, preferentially emerge ORF-first (\autoref{whoisfirst}). With a higher GC-content, ORF length above which RNA-first hypothesis is more feasible than ORF-first hypothesis, increases. For example, this number is 48 codons for a GC-content of 0.45 (data not shown).

A more stringent definition of ORF-first hypothesis would describe a two-step process. In the first step, an ORF emerges in an untranscribed region of DNA, but transcription does not emerge. In the second step transcription emerges, while ORF stays intact (\autoref{orffirsteq2}). Likewise, RNA-first hypothesis can be defined by a two-step probability. In the first step transcription emerges in a non-genic DNA (that does not have an ORF), and in the second step an ORF emerges, while transcription remains intact (\autoref{rnafirsteq2}).

\begin{align}
P_\textit{ORF-first}' & = \underbrace{P_\textit{ORF-gain} \times (1-P_\textit{RNA} -P_\textit{RNA-gain})}_\textit{first step} \times \underbrace{P_\textit{ORF-stay}\times P_\textit{RNA-gain}/(1-P_\textit{RNA})}_\textit{second step} \label{orffirsteq2}\\[1em]
P_\textit{RNA-first}' & = \underbrace{P_\textit{RNA-gain} \times (1-P_\textit{ORF} -P_\textit{ORF-gain})}_\textit{first step} \times \underbrace{P_\textit{RNA-stay}\times P_\textit{ORF-gain}/(1-P_\textit{ORF})}_\textit{second step}\label{rnafirsteq2}
\end{align}

With this stringent definition, the difference between the probabilities of the two trajectories becomes very small ($< 3\times10^{-5}\ \%$). However, we still find that \textit{de novo} genes with shorter ORFs ($<$ 34 codons) have a slightly higher chance of emerging ORF first.

Overall, long ORFs are more likely to be lost before transcription emerges than short ORFs, and hence they preferentially emerge in existing transcripts. Conversely, short ORFs are more likely to stay undisrupted until the emergence of transcription.

\subsection{Would extensive transcription loss suggest negative selection of toxic proteins?}


\textit{De novo} genes that do not provide any fitness benefit to an organism are not fixed in populations via natural selection. These genes can be lost due to mutation pressure. Some newborn \textit{de novo} genes can also encode toxic proteins, that may aggregate or interfere with physiology in some other way. These genes would thus be eliminated from the population genomes via negative selection. We remind the reader that an ORF is lost if the start codon is mutated, the stop codon is mutated to an amino acid encoding codon (non-stop mutation), or if an amino acid encoding codon is mutated to a stop codon (premature stop/non-sense mutation). It is likely that a non-stop mutation or a non-sense mutation, can still result in translation of a protein (extended or truncated, respectively). Furthermore, non-stop mutations can also lead to cellular toxicity \citep{nonstop}. Thus ORF loss does not ensure elimination of toxic proteins, which in turn suggests that transcription loss may be the better way to inactivate the associated genes.

To understand which is the most probable mechanism of gene loss, we compared the probabilities of ORF loss ($P_\textit{ORF-loss}$) and transcription loss ($P_\textit{RNA-loss}$). We found that ORF loss is more probable than transcription loss, for all ORF longer than 33 codons (\autoref{lossprob}).

We made our analysis of gene loss more stringent by comparing the probabilities of two evolutionary scenarios, both of which begin with an intact \textit{de novo} gene. In the first scenario, the ORF remains intact whereas the transcription is lost. The probability of this scenario is described by: 
$P_\textit{ORF-stay} \times P_\textit{RNA-loss}/P_\textit{ORF}$. The probability of the second scenario, where transcription stays intact while ORF is lost, is: $P_\textit{ORF-loss} \times P_\textit{RNA-stay}/P_\textit{RNA}$. We find that the first scenario is more plausible for ORFs that have less than 34 codons, which is in agreement with our previous analysis (\autoref{lossprob}). Also, the probability values of scenario one and two, obtained from this stringent analysis, were not significantly different from the transcription ($P_\textit{RNA-loss}$) and ORF loss ($P_\textit{ORF-loss}$) probabilities, respectively. 

Thus, widespread transcription loss can suggest a negative selection on protein expression.

\begin{figure}[!t]
\centering
\includegraphics[scale=0.6]{Figures/pLoss.pdf}
\caption{ORF loss is more probable than transcription loss for \textit{de novo} genes containing more than 33 codons. The horizontal axis denotes the number of codons in the ORF. The vertical axis shows the log transformed ratio of ORF loss and RNA loss probabilities, such that a positive value means ORF-loss is more probable than RNA loss, and \textit{vice versa}.}
\label{lossprob}

\vspace{1ex}
\hrule
\end{figure}


\subsection{Does mutation bias shape protein composition?}

\begin{figure}[!t]
\centering
\begin{subfigure}{0.47\textwidth}
\caption{}
\flushleft
\includegraphics[scale=0.5]{Figures/pAAfreq.pdf}
\end{subfigure}\hfill%
\begin{subfigure}{0.47\textwidth}
\caption{}
\flushleft
\includegraphics[scale=0.5]{Figures/pToAA.pdf}
\end{subfigure}

\vspace{1em}

\begin{subfigure}[t]{0.47\textwidth}
\caption{}
\flushleft
\includegraphics[scale=0.5]{Figures/pFrAA.pdf}
\end{subfigure}\hfill%
\begin{subfigure}[t]{0.47\textwidth}
\caption{}
\flushright
\includegraphics[scale=0.5]{Figures/pAAsubSymCond.pdf}
\end{subfigure}

\caption{Mutation bias and GC-content shape protein composition. Panels \textbf{(A)}, \textbf{(B)} and \textbf{(C)} show expected frequency, gain probability (\autoref{aagaineq}), and loss probability (\autoref{aalosseq}) for different amino acids (horizontal axes). The vertical axis in all these three subplots show the normalized values of these measures such that their sum over all amino acids is equal to one. Panel \textbf{(D)} shows the directionality (colored tiles, $\textit{D}_{x\to y}$, \autoref{directionality}) of the substitution from an amino acid (vertical axis) to another (horizontal axis). Positive values of $\textit{D}_{x\to y}$ are shown in shades of red, whereas negative values are shown in shades of blue.}
\label{aafreq}

\vspace{1ex}
\hrule
\end{figure}

In previous sections, we showed that mutation bias affects the rate of \textit{de novo} gene emergence and loss. We next turned our attention to whether this bias affects the composition (and thereby the chemistry) of \textit{de novo} proteins. To this end, we first asked if the expected frequency of different amino acids is uniform, and found that it is not uniform. Amino acids like leucine (L) and serine (S) have a higher expected frequency than other amino acids. On the other hand, amino acids like methionine (M) and tryptophan (W) are less probable (\autoref{aafreq}{\color{blue}A}). This non-uniformity is to a great extent determined by number of degenerate codons for an amino acid. However, GC-content also plays a role. For example, leucine, arginine and serine, all have six codons each. However, leucine is more probable than serine which in turn is more probable than arginine (\autoref{aafreq}{\color{blue}A}). 

Next, we aimed to find out if some amino acid substitutions are more probable, than the others. To this end, we calculated the gain probability (\autoref{aagaineq}) for every amino acid. Specifically for every amino acid $x$, we calculated the average probability that it substitutes any of the other nineteen amino acids, due to mutations. More precisely, the total probability that an amino acid $y$ will mutate to an amino acid $x$ is the product of the expected frequency of amino acid $y$ ($P_y$), and the substitution rate from $x$ to $y$ ($\mu_{y\to x}$). The gain probability for an amino acid $x$ is then defined as:

\begin{equation}
G_x = \sum_{b \neq a} P_y \times \mu_{y\to x}
\label{aagaineq}
\end{equation}

We found that amino gain probabilities were also non-uniform across different amino acids (\autoref{aafreq}{\color{blue}B}) but had a good positive correlation with the expected frequency (Pearson $\rho = 0.955, P<10^{-10}$).

Next, we calculated how easily is an amino acid mutated to any other amino acid. Specifically, the conditional loss probability for an amino acid ($L_x$), given the amino acid already exists, is defined as:

\begin{equation}
L_x = \sum_{x \neq y} \mu_{x\to y}
\label{aalosseq}
\end{equation}


The loss probability was also not uniformly distributed (\autoref{aafreq}{\color{blue}C}) but it did not significantly correlate with the expected frequency of the amino acids (Pearson $\rho = -0.255, P = 0.2818$).

Next, we investigated if some amino acid substitutions are more common than the others. As expected, amino acid substitutions that require more than one nucleotide change were much less probable than those needing just a single nucleotide change. Therefore, we focused our analysis on amino acid substitutions whose total probabilities were more than that of a double nucleotide mutation (note that this procedure identifies amino acid substitutions that are possible via single nucleotide mutation, but it does not exclude multinucleotide mutations from the calculation of amino acid substitution probabilities). Although single nucleotide mutations are more probable than double mutations, the different single mutations do not occur at the same rate due to mutation rate bias. Thus an amino acid substitution (for example K$\to$E) may not be as likely to occur as the reverse substitution (E$\to$K). To verify if such a directionality exists, we calculated the log transformed ratio of forward and reverse substitution probability, $D_{x\to y}$, for every pair of amino acids, $x$ and $y$. A positive value of $D_{x\to y}$ means that $x$ to $y$ substitution is more likely than $y$ to $x$ substitution, and \textit{vice versa}.

\begin{equation}
\textit{D}_{x\to y} = \log_2\left(\frac{\mu_{x\to y}}{\mu_{y\to x}}\right)
\label{directionality}
\end{equation}

As we mentioned in previous paragraph, we restricted this analysis to substitutions that can be achieved via single nucleotide mutations. We found that most substitutions indeed had a preferred direction ($\textit{D}_{xy} \neq 0$,  \autoref{aafreq}{\color{blue}D}). The median absolute value of $\textit{D}_{xy}$ was 0.905 and the maximum value was 2.836. That is, in 38 out of 75 amino acid pairs we analysed, one direction of substitution was at least 1.87 times more probable than the other. For example, E$\to$K substitution is 1.8 times more probable than K$\to$E substitution. This may appear as a small number but many such unidirectional substitutions occurring in an evolving protein could indicate a directional selection. We also performed an analogous analysis where we defined the substitution directionality using the ratio of corresponding amino acid gain probabilities (\autoref{aagaineq}). Using this definition reduces the range of directionality values ($\textit{D}_{xy}$) such that the median value is 0.38. Although this more stringent definition reduces the intensity of the directionality, it does not change the qualitative results. 


A revious studies has argued that proteins that haven't been shaped via selection, tend to be hydrophobic because hydrophobic amino acids comprise 40\% of amino acids and can also easily arise in a protein sequence due to mutation pressure \citep{hydrophobicRatchet}. They based these conclusions on mutation rate bias estimated in bacteria \citep{bacteriaMutbias}. We asked if our model also makes similar predictions. To this end, we first calculated the percentage of amino acids in a protein sequence that are expected to be hydrophobic. We classified an amino acid hydrophobic if its hydropathy index \citep{hydropathy} is greater than 0 (i.e A, C, F, I, L, M \& V). Based on this definition and the expected frequency of amino acids, we found that $\sim$39\% of amino acids in a protein sequence are expected to be hydrophobic. This finding is in agreement with the previous studies. Next, we asked if a protein sequence in general tends to be hydrophobic. Specifically, we calculated expected hydropathy ($\bar{\alpha}$, \autoref{expHydropathy}) of a protein sequence, based on expected frequency of different amino acids ($P_x$, \autoref{aafreq}) and their hydropathic index values ($\alpha_x$). 

\begin{equation}
\bar{\alpha} = \sum_{x} P_x \alpha_x
\label{expHydropathy}
\end{equation}


We found that the expected hydropathy is $-0.08628$, which suggests that random protein sequences are not generally hydrophobic. 
 
To find out if random mutations indeed cause hydrobphobic amino acids to occur in protein sequences, we calculated the probability that any mutation substitutes a non-hydrophobic amino acid with a hydrophobic amino acid ($P_{\alpha\textit{-gain}}$, \autoref{alphagain}). Likewise, we calculated the probability that any mutation causes a hydrophobic to non-hydrophobic amino acid substitution ($P_{\alpha\textit{-loss}}$, \autoref{alphaloss}).  

\begin{align}
P_{\alpha\textit{-gain}} & = \sum_{\genfrac{}{}{0pt}{1}{x}{\,|\,\alpha_x>0}} \sum_{\genfrac{}{}{0pt}{1}{y}{\,|\,\alpha_y \geq 0}} P_y \times \mu_{y\to x}\label{alphagain}\\[1em]
%
P_{\alpha\textit{-loss}} &= \sum_{\genfrac{}{}{0pt}{1}{x}{\,|\,\alpha_x>0}} \sum_{\genfrac{}{}{0pt}{1}{y}{\,|\,\alpha_y \geq 0}} P_x \times \mu_{x\to y}\label{alphaloss}
\end{align}

Next, we calculated the ratio of $P_{\alpha\textit{-gain}}$ and $P_{\alpha\textit{-loss}}$, and found that it was 1.053. This shows that random mutations do not preferentially cause hydrophobic amino acids to accumulate in protein sequence. Inclusion of proline and tryptophan in the list of hydrophobic amino acids does not change this finding.

\section{Discussion} 

In this work, we addressed some fundamental questions about \textit{de novo} gene evolution. Broadly, we asked how quickly these genes can emerge and be lost, whether their birth and death has a preferred trajectory, and how the composition of their protein sequence is determined. To this end, we developed mathematical models and used them to address specific questions. These models are based on mutation probability and mutation rate bias, and represent neutral evolution where DNA sequences are not evolving under natural selection. Thus our models provide an opportunity to test the alternative hypothesis that some \textit{de novo} genes are evolving under selection.

We show that genes can be lost much more rapidly than they are gained, such that a gene can be independently lost two to three times in the expected time to gain it. Therefore, presence of a \textit{de novo} gene in one species (or population) cannot be interpreted as an evidence of gene emergence in this species.

We answer the long standing question of what is the sequence of \textit{de novo} emergence. That is, whether ORF emerges first or transcription emerges first \citep{EBB-F1000}. We show that, in the absence of selection, long ORFs are likely to emerge in existing transcripts (transcription emerges first), whereas short ORFs are likely to emerge before transcription emerges in the genomic locus. Long non-coding RNAs may be a potential source of \textit{de novo} genes with long ORFs, but this evolutionary process may be constrained by a different set of factors. A previous study has shown that new transcripts frequently arise in regions overlapping existing genes. These new transcripts can harbor ORFs which in turn leads to \textit{de novo} gene emergence \citep{albaYeastdenovo}. This is an example of RNA-first trajectory. However, the ORFs present in these genes may be under indirect selection pressure that acts on the overlapping gene sequences. Therefore, they may be less frequently mutated or lost.

Although our model is based on the assumption of neutral evolution, one can speculate the effects of positive selection on the \textit{de novo} gene. For example, if the gene produces a protein that is beneficial to the organism, then selection would cause the fixation of the gene as soon as the ORF emerges via RNA-first trajectory. On the other hand, the selection would have no effect on an ORF that is not transcribed. This is based on the assumption that the ORF sequence has no other function (for e.g. regulation), or is not under indirect selection (for e.g. if it overlaps with an existing gene). Thus positive selection will intensify the preference for RNA-first trajectory.

We show that transcription loss is less likely than the loss of long ORFs, However, it may be difficult to find direct evidence for such a phenomenon. Specifically, it is difficult to infer from data if transcription was lost or it never emerged. If the protein encoded by an ORF is indeed toxic, it would be silenced rapidly in microevolutionary time scales. One way of inferring negative selection from transcription loss may involve correlation of ORF sequence variation and transcription status. Mutations within ORF sequence are much more probable than mutations that disrupt transcription or ORF. This is so because the entire ORF sequence has higher number of mutable sites. If an ORF variant is untranscribed in many populations (species or taxa), then it is possibly toxic.  

Finally, we show that some amino acids substitutions are more likely to occur than the others, and frequent  substitutions that are expected to occur less frequently under neutral model, can suggest directional evolution. 

Like all theoretical models, our models are not without limitations. First, we assume that nucleotides are uniformly distributed according to GC-content. This is not true as nucleotide composition can vary significantly across the genome. Mutation rates are also not uniform across the genome and can vary significantly \citep{mutbiasArabidopsis}. However, our models can be adapted to investigate specific genomic loci, where information on the local GC-content and mutation rate can more accurately determine the \textit{de novo} gene evolution in these loci. Second, our models are based on parameters such as mutation rate and mutation bias, which we obtained from published data on human and \textit{D.melanogaster} genomes \citep{humanmutrate,drosophilamutrate}. The accuracy of the models' prediction will depend on the accuracy of parameters. However, mutation rate itself only determines the values of different probabilities. Our results, which are based on probability ratios, will not significantly change with a different value of mutation rate. More importantly, different species may have a different spectrum of mutations \citep{joshmutbias}, especially so for species that are highly distant from each other. Thus our inferences based on our chosen parameters will not universally apply to all species. Third, we assume that all mutations are independent. This is not the case for tandem mutations, where multinucleotide mutations occur can more frequently than their expected frequency under the assumption of mutational independence \citep{MNM}. Finally, we ignore insertions, deletions and transpositions as mechanisms of mutation. Insertions and deletions (indels) are reported to be less frequent than substitutions \citep{drosophilamutrate}, and both indels transpositions cause larger sequence alterations than point mutations. However, our modeling methodology allows incorporation of these mechanisms as a source of mutations. 

A famous aphorism about theoretical models says ``all models are wrong but some are useful'' \citep{GEBox}. This is true for our models too. They are not 100\% accurate but they are useful for making predictions that can lead a more focused experimental validation. These models can be made more complex to accommodate diverse molecular mechanisms driving gene evolution. Therefore our work opens up an opportunity for theoretical and computational biologists to develop models using our modeling framework, that more accurately describe their system of interest.

\section{Materials and Methods}

\subsection{Mutation probabilities}

We calculated nucleotide substitution probabilities based on mutation rate and mutation rate bias data. Specifically, we  used a mutation rate ($\lambda$) of $7.8\times10^{-9}$ mutations per nucleotide position per generation \citep{drosophilamutrate}. We derived our mutation bias parameters from two published studies, the first on \textit{Drosophila melanogaster} \citep{drosophilamutrate}, and the second on humans \citep{humanmutrate}. \autoref{mutbias} shows the exact values of mutation bias probabilities that we used in this study.

\begin{table}[H]
\centering
\begin{tabular}{c c}
\toprule
\textbf{Substitution} & Probability($\mu$) \\\midrule
\texttt{A:T}$\to$\texttt{T:A} & 0.056 \\\midrule
\texttt{A:T}$\to$\texttt{G:C} & 0.243 \\\midrule
\texttt{A:T}$\to$\texttt{C:G} & 0.074 \\\midrule
\texttt{G:C}$\to$\texttt{A:T} & 0.483 \\\midrule
\texttt{G:C}$\to$\texttt{T:A} & 0.075 \\\midrule
\texttt{G:C}$\to$\texttt{C:G} & 0.069 \\\bottomrule
\end{tabular}
\caption{Mutation bias probabilities.}
\label{mutbias}
\end{table}

\subsection{Probabilities of finding, gaining, and losing a DNA sequence}
\label{methbasic}

\subsubsection{Probability of finding a DNA sequence}
\label{methprob}

We calculated the probability of finding a DNA sequence based on global nucleotide frequency distributions, given by the GC-content. We used a GC-content of 0.42 for our calculations. Specifically, the probability of finding either a \texttt{G} or a \texttt{C} is: $S = 0.5\times$GC-content. The probability of finding an \texttt{A} or a \texttt{T} is: $W = 0.5 - S$. Using these values, we calculated the probability of finding a DNA sequence motif by chance. For example, the probability of finding the sequence \texttt{ATG} would be: $W\times W \times S$.

\subsubsection{Probability of gaining a DNA sequence}
\label{methgain}

We calculated the probability of gaining a DNA sequence due to mutations using GC-content, mutation rate and mutation bias. Specifically, we calculated the probability that a DNA sequence does not exist, and it emerges due to specific nucleotide mutations. More precisely, this probability is the product of two other probabilities. The first is the probability of finding a DNA sequence ($x$) that is not the sequence of interest. The second probability is that this sequence $x$ mutates to the sequence of interest. To explain this calculation better, we use the example of \texttt{CTA} mutating to \texttt{ATG}. The first probability i.e. the probability of finding \texttt{CTA} by chance is $SW^2$. \texttt{CTA} mutates to \texttt{ATG} via two nucleotide mutations (\texttt{C}$\to$\texttt{A} and \texttt{A}$\to$\texttt{G}). Thus the probability of this DNA change would be the probability of two nucleotide mutations ($\lambda^2$) multiplied by two mutation bias probabilities (\texttt{G:C}$\to$\texttt{T:A} and \texttt{A:T}$\to$\texttt{G:C}). Overall, the chance of \texttt{CTA} mutating to \texttt{ATG} would be:

$$SW^2 \times \lambda^2 \times \mu_{\texttt{G:C}\to\texttt{T:A}} \times \mu_{\texttt{A:T}\to\texttt{G:C}}$$

Next, we calculated the probability that every nucleotide triplet that is not \texttt{ATG}, mutates to \texttt{ATG}. This can happen via one, two, or three nucleotide mutations. The sum of all these mutation probabilities is the probability of \texttt{ATG} gain.

Using the same principle we calculated the gain probability of any DNA sequence motif (of any length or composition). We excluded insertions and deletions as a mechanism of gain of small DNA sequences that we analysed in this study. 

\subsubsection{Probability of losing a DNA sequence}

We calculated the loss of a DNA sequence motif using the same principle we used for calculating gain probabilities. However, we defined loss probability as a conditional probability i.e. we assume that the DNA sequence of interest already exists in the genome. For example, the loss probability of a specific \texttt{ATG} sequence would be the sum of probabilities of \texttt{ATG} mutating to any of the other 63 nucleotide triplets (via one, two, or three nucleotide mutations). We use conditional loss probabilities by default, because usually one is interested in finding out how quickly can an existing DNA sequence erode.

We used this method to calculate the loss probability of any DNA sequence motif, and we excluded insertions and deletions from this calculation.

\subsection{Probabilities of finding, gaining, and losing DNA features}
\label{methfeatures}

Usually, a specific function is encoded in DNA by several DNA sequences. For example, translation stop is encoded by three codons (\texttt{TGA}, \texttt{TAG}, \texttt{TAA}). We use the term DNA features to mean a set of DNA sequences that are associated with the same function. For every such DNA feature set, there is a complementary set of non-features i.e DNA sequences that are not associated with the feature's function. For example, the non-feature set of stop codons would be all the other 61 codons. 

The probability that a DNA feature exists, is the sum of probabilities of every DNA sequence in that set (\autoref{methprob}).

The probability that a DNA feature is gained via mutations, is the sum of probabilities of every non-feature sequence mutating to any feature sequence. If $F$ denotes the feature set, and $\mu_{y\to x}$ denotes the probability of a DNA sequence $y$ mutating to a DNA sequence $x$ (see \autoref{methgain}), then:

\begin{equation}
P_\textit{feature-gain} = \sum_{x \in F} \sum_{y \notin F} P_y \times \mu_{y\to x}
\end{equation}

The probability that a DNA feature is lost via mutations is a conditional probability that given a feature exists, it mutates to any of the non-feature sequences. 

\begin{equation}
P_\textit{feature-loss} = \frac{\displaystyle\sum_{x \in F} \sum_{y \notin F} P_x \times \mu_{x\to y}}{\displaystyle\sum_{x \in F} P_x}
\end{equation}

Because a feature set usually has many DNA sequences, a mutation can change a feature sequence such that the resulting sequence is also a part of the feature set. Thus we defined the probability ($P_\textit{feature-stay}$) that a feature does not erode because of mutations. Specifically, it is the the sum of two probabilities. First is the probability that no mutation occurs in the DNA sequence ($P_0$), and the second probability describes the event where mutated sequence is still a part of the feature set.

\begin{equation}
P_\textit{feature-stay} = P_0 + \sum_{x \in F} \sum_{\genfrac{}{}{0pt}{1}{y \in F}{y\neq x}} \mu_{x\to y}
\end{equation}


The probability that no mutation occurs ($P_0$) in a DNA sequence of length $k$ is described by Poisson distribution. 

$$P_0 = 1-e^{-k\lambda}$$

Because the mutation rate is biased (\autoref{mutbias}), the probability that no mutation occurs in a DNA sequence depends on its composition. 

The probability that an \texttt{A} or a \texttt{T} mutates ($\lambda_\texttt{AT}$), is thus described as:

$$\lambda_\texttt{AT} = 2\times\lambda\times(\mu_{\texttt{A:T}\to\texttt{T:A}} + \mu_{\texttt{A:T}\to\texttt{G:C}} + \mu_{\texttt{A:T}\to\texttt{C:G}})$$

Likewise, the probability that a \texttt{G} or a \texttt{C} mutates ($\lambda_\texttt{GC}$) is: 

\vspace{-1ex}

$$\lambda_\texttt{GC} = 2\times\lambda\times(\mu_{\texttt{G:C}\to\texttt{A:T}} + \mu_{\texttt{G:C}\to\texttt{T:A}} + \mu_{\texttt{G:C}\to\texttt{C:G}})$$

(Note that the general mutation rate, $\lambda$, is an average of $\lambda_\texttt{AT}$ and $\lambda_\texttt{GC}$.)

\vspace{1\baselineskip}

Thus the probability that a sequence of length $k$, containing $m$ number of \texttt{A} and \texttt{T}, does not mutate is:

\vspace{-1ex}

$$P_0 = (1-e^{-m\lambda_\texttt{AT}})\times(1-e^{-(k-m)\lambda_\texttt{GC}})$$

\subsection{Probabilities of finding, gaining, and losing transcription}

To calculate the probabilities of finding, gaining, and losing transcription, we first defined the minimum requirements for transcription. These requirements are a promoter, either a TATA-box or an Initiator (Inr) element, and a polyadenylation signal. Each of these three DNA sequence motifs consists of more than one DNA sequence.

\subsubsection{Definition of TATA-box, Inr and poly-A signal}

\paragraph{TATA box:}

The consensus sequence of TATA-box is \texttt{TATA [T/A] A [T/A] [G/A]} \citep{tata2} such that, any of the nucleotides enclosed within the square brackets, can exist in that specific position in the sequence. Another study used X-ray crystallography to identify mutations that are tolerated in TATA-box consensus sequence \citep{tata1}. Thus there are eight consensus sequences, and at seven positions, a non-consensus nucleotide is tolerated (\autoref{tatatab}). This makes a TATA-box feature set that contains 104 sequences. We defined the corresponding non-feature set by generating all 8-mer DNA sequences that do not belong to the TATA-box feature set.

\begin{table}[!t]
\centering
\begin{tabular}{c c}
\toprule
Position  & Accepted nucleotides \\ \midrule
1         & T$\gg$C$>$A,G        \\ \midrule
2         & A$\gg$T               \\ \midrule
3         & T$\gg$A,C             \\ \midrule
4         & A$\gg$T               \\ \midrule
5         & T/A                   \\ \midrule
6         & A$\gg$G$>$C/T         \\ \midrule
7         & A/T$>$G$>$C           \\ \midrule
8         & G/A$>$C/T             \\ \bottomrule
\end{tabular}
\caption{Nucleotide residues that are tolerated at different positions in a TATA-box \citep[based on][]{tata1}}
\label{tatatab}
\end{table}


\paragraph{Inr:} The consensus sequence of Inr is \texttt{[T/G/C] [T/G/C] CA [T/G/C] [T/A]}, which gives 54 sequences in the Inr feature set \citep{Inr}. We generated the corresponding non-feature set with all DNA 6-mers that are distinct from the 54 Inr sequences.

\paragraph{poly-A signal:} There are four known poly-A signal variants -- \texttt{AATAAA}, \texttt{ATTAAA}, \texttt{AGTAAA}, and \texttt{TATAAA} that compose the poly-A signal feature set \citep{polyA}. The non-feature set consists of all other DNA 6-mers.

\vspace{1\baselineskip}

We calculated probabilities of finding, gaining and losing these features as we described in the previous section.

\subsubsection{Definition of the probabilities}

Probability that a DNA region is transcribed by chance ($P_\textit{RNA}$), is the probability of finding a poly-A signal multiplied by the sum of probabilities of finding a TATA-box, and an Inr element. 

\begin{equation}
P_\textit{RNA} = P_\textit{polyA} \times (P_\textit{TATA} + P_\textit{Inr})
\label{eqrnaprob}
\end{equation}

Gain of transcription can happen via two mechanisms. In the first mechanism a promoter (TATA-box or Inr) already exists, and mutations cause a poly-A signal to emerge ($P_\textit{polyA-gain}$) but do not destroy the promoter ($P_\textit{TATA-stay}$,  $P_\textit{Inr-stay}$). In the second mechanism, a poly-A signal exists and the promoter emerges due to mutations ($P_\textit{TATA-gain}$,  $P_\textit{Inr-gain}$), while the poly-A signal remains undisrupted ($P_\textit{polyA-stay}$). Thus the total probability of transcription gain ($P_\textit{RNA-gain}$) would be the sum of the probabilities of the two mechanisms.

\begin{equation}
P_\textit{RNA-gain} = (P_\textit{TATA-stay} + P_\textit{Inr-stay})\times P_\textit{polyA-gain} + (P_\textit{TATA-gain} + P_\textit{Inr-gain})\times P_\textit{polyA-stay}
\label{eqrnagain}
\end{equation}

It is also possible that both the promoter and the poly-A signal were initially absent, and emerge via mutations. However, this probability is very small and is negligible.

Transcription is lost when either the promoter is lost or when the poly-A signal is lost. Since two kinds of promoters can initiate transcription, there can be three kinds of transcription units. First that has only TATA-box, second that has only Inr, and the third that has both these promoters. All these three transcription units must also have a poly-A signal. To calculate the conditional probability of promoter loss, we calculated the fraction of transcription units ($Q$) that belong to each of these three classes. For example, the fraction of transcription units that are TATA-only is ($Q_\textit{TATA-only}$):

\begin{equation*}
Q_\textit{TATA-only} = \frac{P_\textit{TATA}\times(1 - P_\textit{Inr})}{(P_\textit{TATA} + P_\textit{Inr} - P_\textit{TATA}\times P_\textit{Inr})}
\end{equation*}

\vspace{1ex}

Here, the numerator denotes probability of finding a TATA-box but not an Inr, and the denominator denotes the probability of finding a promoter that can be a TATA-box, Inr, or both. Likewise, the fraction of Inr-only ($Q_\textit{Inr-only}$) and TATA-Inr promoters ($Q_\textit{both}$) are given by:

\begin{align*}
Q_\textit{Inr-only} & = \frac{P_\textit{Inr}\times(1 - P_\textit{Inr})}{(P_\textit{TATA} + P_\textit{Inr} - P_\textit{TATA}\times P_\textit{Inr})} \\[1ex]
Q_\textit{both} & = 1 - Q_\textit{TATA-only} -Q_\textit{Inr-only}
\end{align*}

For a TATA-only transcription unit to lose a promoter, it should lose the TATA-box but simultaneously not gain an Inr. Likewise, an Inr-only transcription unit will lose a promoter when it loses the Inr but not gain a TATA-box. A transcription unit that has both the promoters must lose both of them. For all the three kinds of transcription units, transcription would be lost if poly-A signal is lost.

The conditional probability of transcription loss ($P_\textit{RNA-loss}$), given that transcription exists, is thus defined as:

\begin{align}
P_\textit{RNA-loss} & = Q_\textit{TATA-only} \times P_\textit{TATA-loss}\times(1-P_\textit{Inr-gain}) \nonumber\\[1pt]
 & \quad + Q_\textit{Inr-only}\times P_\textit{Inr-loss}\times(1-P_\textit{TATA-gain}) \nonumber \\[1pt]
 & \quad + Q_\textit{both}\times P_\textit{TATA-loss} \times P_\textit{Inr-loss} + P_\textit{polyA-loss}
\label{eqrnaloss}
\end{align}

\subsubsection{Probability that transcription remains intact}

Transcription remains intact if neither the promoter, nor the poly-A signal is lost. More specifically, we define this probability ($P_\textit{RNA-stay}$) as:

\begin{equation}
P_\textit{RNA-stay} = (P_\textit{TATA-stay} + P_\textit{Inr-stay}) \times P_\textit{polyA-stay}
\label{eqrnastay}
\end{equation}

\subsection{Probabilities of finding, gaining, and losing an ORF}

\subsubsection{Probability of finding an ORF}

An open reading frame (ORF) is a nucleotide sequence with a length that is a multiple of three, that begins with a start codon (\texttt{ATG}), and ends with one of the three stop codons (\texttt{TAG}, \texttt{TGA}, \texttt{TAA}). This necessarily means that there are no stop codons within the sequence. Thus the probability of finding an ORF containing $k$ codons including start and stop codons ($P_\textit{ORF}$) is: 

\begin{equation}
P_\textit{ORF} = P_\textit{ATG} \times P_\textit{stop} \times (1 - P_\textit{stop})^{k-2}
\label{eqorfprob}
\end{equation}

Here, $P_\textit{ATG}$ and $P_\textit{stop}$ are the probabilities of finding a start codon, and a stop codon by chance, respectively.

\subsubsection{Probability of gaining an ORF}

As we defined in the previous section, an ORF has three requirements (start codon, stop codon, and no premature stop codon in the sequence). Thus an ORF can emerge due to mutations via three mechanisms. In each of these mechanisms, one requirement is initially absent whereas the other two are present. Then mutations cause the missing requirement to emerge while not disrupting the other two requirements. More specifically, the ORF can be gained via the following three mechanisms:
\begin{enumerate}
\item Gain of a start codon ($P_\textit{ATG-gain}$) while a stop codon continues to exist at the end of a reading frame ($P_\textit{stop-stay}$), and there is no emergence of stop codon within the reading frame ($1- P_\textit{stop} - P_\textit{stop-gain}$).
\item Gain of a stop codon ($P_\textit{stop-gain}$), while a start codon continues to exist at the beginning of a reading frame ($P_\textit{ATG-stay}$), and there is no emergence of stop codon within the reading frame.
\item Loss of a premature stop codon, at any of the $k-2$ codon positions within the reading frame ($P_\textit{stop-gain}$). At the same time start and stop codons remain undisrupted by mutations, and no stop codon emerges at any of the other $k-3$ positions.
\end{enumerate} 

Thus we define the probability of ORF gain ($P_\textit{ORF-gain}$) as:

\begin{align}
P_\textit{ORF-gain} & = \quad P_\textit{ATG-gain}\times P_\textit{stop-stay} \times (1- P_\textit{stop} - P_\textit{stop-gain})^{k-2} \nonumber \\[1pt]
& \quad + P_\textit{ATG-stay}\times P_\textit{stop-gain} \times (1- P_\textit{stop} - P_\textit{stop-gain})^{k-2} \nonumber \\[1pt]
& \quad + P_\textit{ATG-stay}\times P_\textit{stop-stay} \times P_\textit{stop-loss}\times(k-2) \times (1- P_\textit{stop} - P_\textit{stop-gain})^{k-3} 
\label{eqorfgain}
\end{align}

\subsubsection{Probability of ORF loss}

ORF can be lost when any of its three requirements are lost. We thus define the conditional probability of ORF loss as:

\begin{equation}
P_\textit{ORF-loss} = P_\textit{ATG-loss} + P_\textit{stop-loss} + (k-2)\times \frac{P_\textit{stop-gain}}{1-P_\textit{stop}}
\label{eqorfloss}
\end{equation}

The last term in this equation describes the conditional probability of stop-gain, given the assumption that no stop codon exists within the ORF.

\subsubsection{Probability that ORF remains intact}

We assumed that an ORF of a certain length remains intact if none of the necessary features are lost. However, the ORF sequence can mutate to cause non-synonymous changes in the translated protein sequence. This condition applies to all the three ORF probabilities described above. We define the probability that an ORF remains intact ($P_\textit{ORF-stay}$) as:

\begin{equation}
P_\textit{ORF-stay} = P_\textit{ATG-stay} \times P_\textit{stop-stay} \times (1 - P_\textit{stop} - P_\textit{stop-stay})^{k-2}
\label{eqorfstay}
\end{equation}

\subsection{Probability of amino acid substitutions}

We calculated the probability of amino acid substitutions based on nucleotide substitution in the corresponding codons. In this case, multiple codons that code for the same amino acid constitute a feature set (\autoref{methfeatures}). Similarly when analysing hydrobhobic to non-hydrophobic substitutions, all codons that represent hydrophobic amino acids form a feature set (and \textit{vice versa}).

\section{Data availability}
We performed all calculations using Julia programming language, and the all scripts are freely available on GitHub: BharatRaviIyengar/DeNovoEvolution. 

\bibliographystyle{plainnat}

\small
\bibliography{refs}

\end{document}
